ee/maintained-apps/ingesters/homebrew/external_refs/microsoft_version_from_release_notes.go

```diff
@@ -0,0 +1,47 @@
+package externalrefs
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	macoffice "github.com/fleetdm/fleet/v4/server/vulnerabilities/macoffice"
+)
+
+var (
+	releaseNotesCache macoffice.ReleaseNotes
+	cacheOnce         sync.Once
+)
+
+// MicrosoftVersionFromReleaseNotes returns the short version from the release notes given a Homebrew version.
+// Homebrew version "16.95.25032931"
+// Release notes version "Version 16.95.3 (Build 25032931)"
+// For this example it would return "16.95.3"
+func MicrosoftVersionFromReleaseNotes(args ...interface{}) (string, error) {
+	if len(args) > 0 {
+		if homebrewVersion, ok := args[0].(string); ok {
+			versionParts := strings.Split(homebrewVersion, ".")              // homebrew version format is like "16.95.25032931"
+			version := strings.Join(versionParts[:len(versionParts)-1], ".") // Extract version without the build number
+			build := versionParts[len(versionParts)-1]                       // Extract the build number
+
+			var err error
+			cacheOnce.Do(func() {
```

Comment:
Why would we benefit from caching here? This isn't a long-running script. Is this for quicker test runs?

Reply:
`MicrosoftVersionFromReleaseNotes` gets called for two attributes, word and excel, so it would make the http call twice. Small optimization... but as you called out, it does help optimize the tests as well. Figured the change to add cache was small and simple enough to add.

Reply:
Got it. Fair enough.